src/EditorFeatures/Test2/GoToDefinition/CSharpGoToDefinitionTests.vb

```diff
@@ -3977,6 +3977,93 @@ public partial class Program
 }", Function(root) root.DescendantNodes().OfType(Of InvocationExpressionSyntax))
         End Function
 
+        <WpfFact>
+        Public Async Function TestInterceptors_FromCallSite1() As Task
+            Await TestInterceptorFromCallSite("
+class C
+{
+    void M()
+    {
+        $$Goo();
+    }
+
+    void [|Goo|]() { }
+}", methodName:="Goo", Function(root) root.DescendantNodes().OfType(Of InvocationExpressionSyntax))
+        End Function
+
+        <WpfFact>
+        Public Async Function TestInterceptors_FromCallSite2() As Task
+            Await TestInterceptorFromCallSite("
+class C
+{
+    void M()
+    {
+        this.$$Goo();
+    }
+
+    void [|Goo|]() { }
+}", methodName:="Goo", Function(root) root.DescendantNodes().OfType(Of InvocationExpressionSyntax))
+        End Function
+
+        <WpfFact>
+        Public Async Function TestInterceptors_FromCallSite3() As Task
+            Await TestInterceptorFromCallSite("
+class C
+{
+    void M(C? c)
+    {
+        c?.$$Goo();
+    }
+
+    void [|Goo|]() { }
+}", methodName:="Goo", Function(root) root.DescendantNodes().OfType(Of InvocationExpressionSyntax))
+        End Function
+
+        Private Async Function TestInterceptorFromCallSite(code As String, methodName As String, getInvocations As Func(Of SyntaxNode, IEnumerable(Of InvocationExpressionSyntax))) As Task
```

Comment:
Is the expecatation that some test will want to use some different kind of getInvocations filter?

Reply:
Taht was the intent.  I didn't take advantage of that.  but i don't mind the code not hardcoding this in, to indicate taht it is up to the test to decide.